mne_bids/tests/test_mne_bids.py

```diff
@@ -249,7 +284,72 @@ def test_cnt():
+
+    event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
+                'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
+    events_fname = op.join(data_path, 'MEG', 'sample',
+                           'sample_audvis_trunc_raw-eve.fif')
+
+    raw = mne.io.read_raw_fif(raw_fname)
+
+    raw_to_bids(subject_id=subject_id, session_id=session_id, run=run,
+                task=task, acquisition=acq, raw_file=raw,
+                events_data=events_fname, output_path=output_path,
+                event_id=event_id, overwrite=False)
+
+    ses_path = make_bids_folders(subject=subject_id, session=session_id,
+                                 root=output_path, make_dir=False)
+
+    data_path = op.join(ses_path, 'meg')
+
+    participants_fname = make_bids_filename(prefix=output_path,
+                                            suffix='participants.tsv')
+    coordsystem_fname = make_bids_filename(
+        subject=subject_id, session=session_id, acquisition=acq,
+        suffix='coordsystem.json', prefix=data_path)
+    data_meta_fname = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        acquisition=acq, suffix='%s.json' % 'meg', prefix=data_path)
+    events_tsv_fname = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        acquisition=acq, suffix='events.tsv', prefix=data_path)
+    channels_fname = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        acquisition=acq, suffix='channels.tsv', prefix=data_path)
+    scans_fname = make_bids_filename(
+        subject=subject_id, session=session_id, suffix='scans.tsv',
+        prefix=ses_path)
+    raw_file_bids = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        acquisition=acq, suffix='%s%s' % ('meg', '.fif'))
+
+    with pytest.raises(OSError, match="already exists"):
+        _channels_tsv(raw, channels_fname, overwrite=False, verbose=False)
+    with pytest.raises(OSError, match="already exists"):
+        _sidecar_json(raw, task, 'Elekta', data_meta_fname, 'meg',
+                      overwrite=False, verbose=False)
+    with pytest.raises(OSError, match="already exists"):
+        _participants_tsv(raw, subject_id, "n/a", participants_fname,
+                          overwrite=False, verbose=False)
+    with pytest.raises(OSError, match="already exists"):
+        _coordsystem_json(raw, "n/a", "n/a", 'Elekta', coordsystem_fname,
+                          overwrite=False, verbose=False)
+    with pytest.raises(OSError, match="already exists"):
+        events = _read_events(events_fname, raw)
+        _events_tsv(events, raw, events_tsv_fname, event_id, overwrite=False,
+                    verbose=False)
+    with pytest.raises(OSError, match="already exists"):
+        _scans_tsv(raw, op.join('meg', raw_file_bids), scans_fname,
+                   overwrite=False, verbose=False)
```

Comment:
@monkeyman192 it feels like an overkill to test each of these private functions separately. We only care about the first error that pops up. Also, since they all use `_write_json` and `_write_tsv`, if one works -- they all should work.

Reply:
yeah, I was thinking that this can be removed. It was appropriated before we restructured everything into the `_write_tsv` and `_write_json` functions. Was planning on removing or at least reducing it heavily.